### ISSUE TITLE
Dates before the unix epoch do not parse in the generated oData clients

### ISSUE THREAD

**Issue by carlonnheim** [SAP/cloud-sdk-js#2589]
**Describe the bug**
The payload converter for Edm DateTime values does not allow negative values. The API fails to handle dates earlier than the unix epoch (1970-01-01) due to this.

**To Reproduce**
Generate an oData client and run a get-query to an instance with a date property earlier than 1970-01-01.

**Expected behavior**
The date is parsed properly.

**Screenshots**
Not applicable.

**Used Versions:**

- node version v16.15.0
- npm version 8.5.5
- SAP Cloud SDK version 1.54.2 (used in a CAP project and CAP still uses v1)

**Code Examples**
The issue is in odata-v2/payload-value-converter.js
```
/**
 * @hidden
 * This function can be used for both Edm.DateTime and and Edm.DateTimeOffset.
 */
function edmDateTimeToMoment(edmDateTime) {
    var _a;
    var dateTimeOffsetComponents = (_a = /^\/Date\((?<ticks>\d+)((?<sign>[+-])(?<offset>\d{4}))?\)\/$/.exec(edmDateTime)) === null || _a === void 0 ? void 0 : _a.groups;
    if (!dateTimeOffsetComponents) {
        throw new Error("Failed to parse edmDateTime: ".concat(edmDateTime, " to moment."));
    }
    var timestamp = (0, moment_1.default)(parseInt(dateTimeOffsetComponents.ticks));
    if (dateTimeOffsetComponents.sign && dateTimeOffsetComponents.offset) {
        var offsetMultiplier = dateTimeOffsetComponents.sign === '+' ? 1 : -1;
        var offsetInMinutes = parseInt(dateTimeOffsetComponents.offset);
        return timestamp.utc().utcOffset(offsetMultiplier * offsetInMinutes);
    }
    return timestamp;
}
```
The regex needs to allow a sign on the ticks value, like so
```
    var dateTimeOffsetComponents = (_a = /^\/Date\((?<ticks>[+-]?\d+)((?<sign>[+-])(?<offset>\d{4}))?\)\/$/.exec(edmDateTime)) === null || _a === void 0 ? void 0 : _a.groups;
```

**Log file**
Not applicable

**Impact / Priority**

Affected development phase: UAT

Impact: Impaired

Timeline: Go-Live is in 3 months.

**Additional context**
The same issue seems to be present in v2 as well. We need the fix in v1 since we are using it in a CAP project, which in turn works with v1 still. Thanks!

**Comment by FrankEssenberger** [comment]
Good morning @carlonnheim,

thanks for reporting the issue and also investigating the problem so well. I checked and the issue is also present in version 2.0 and made a quick [PR for that](https://github.com/SAP/cloud-sdk-js/pull/2593). Once this is merged it will be available as `canary` immediately or after the next release which we do every two weeks.

As I understood you are using the SDK via CAP which is still on version 1, but they plan to update. Their plan was to do the update end of June. However, I do not want to promise something for another team. Perhaps there are delays. 
We only do security relevant fixes in version 1 of the SDK which is here clearly not the case, so I hope the colleagues do the update.

In case this is super urgent, you could also try to install the latest SDK version and try to make you local CAP version use it. However, since there a quite a few breaking API changes this is cumbersome.

Best
Frank

**Comment by carlonnheim** [comment]
Thanks for the quick response @FrankEssenberger. We can work with a "patch-package" here until CAP upgrades and we can use the fix from your patch.

**Comment by FrankEssenberger** [comment]
So the PR is merged and a new `canary` version should appear in the next minutes on npm. With this you can try out the the patch or you wait until CAP released a new major version using SDK 2.0. Since it was such a small issue and the tests look fine I would close the issue for now. Feel free to re-open in case the error persists.

Best
Frank
